week-3/slideshow/slideshow.js

```diff
@@ -1,8 +1,131 @@
+
+image.before(imgContainer);
+imgContainer.appendChild(image);
+
+const btnContainer = document.createElement("div");
+btnContainer.classList.add("btn-container");
+
+const backwardBtn = document.getElementById("backward-btn");
+const forwardBtn = document.getElementById("forward-btn");
+
+const stopBtn = document.getElementById("stop");
+
+const autoBackwardBtn = document.getElementById("auto-backward");
+const autoForwardBtn = document.getElementById("auto-forward");
+
+btnContainer.append(
+  autoBackwardBtn,
+  backwardBtn,
+  stopBtn,
+  forwardBtn,
+  autoForwardBtn
+);
+
+let autoPlayBackward;
+let autoPlayForward;
+
+// Write your code here
+
+function generatePhoto() {
+  image.setAttribute("src", images[currentPhotoIndex]);
+  if (currentPhotoIndex === 0) {
+    document.getElementById("img0").style.backgroundColor = "grey";
+    document.getElementById("img1").style.backgroundColor = "black";
+    document.getElementById("img2").style.backgroundColor = "black";
+  } else if (currentPhotoIndex === 1) {
+    document.getElementById("img0").style.backgroundColor = "black";
+    document.getElementById("img1").style.backgroundColor = "grey";
+    document.getElementById("img2").style.backgroundColor = "black";
+  } else if (currentPhotoIndex === 2) {
+    document.getElementById("img0").style.backgroundColor = "black";
+    document.getElementById("img1").style.backgroundColor = "black";
+    document.getElementById("img2").style.backgroundColor = "grey";
+  }
+}
+
+function nextPhoto() {
+  ++currentPhotoIndex;
+  if (currentPhotoIndex === 3) {
+    currentPhotoIndex = 0;
+  }
+  generatePhoto();
+}
+
+function previousPhoto() {
+  --currentPhotoIndex;
+  if (currentPhotoIndex === -1) {
+    currentPhotoIndex = 2;
+  }
+  generatePhoto();
+}
+
+backwardBtn.addEventListener("click", previousPhoto);
+forwardBtn.addEventListener("click", nextPhoto);
+
+autoBackwardBtn.addEventListener("click", () => {
+  autoPlayBackward = setInterval(() => {
+    previousPhoto();
+  }, 2000);
+  clearInterval(autoPlayForward);
+  autoBackwardBtn.disabled = true;
+  autoForwardBtn.disabled = true;
+});
+
+autoForwardBtn.addEventListener("click", () => {
+  autoPlayForward = setInterval(() => {
+    nextPhoto();
+  }, 2000);
+  clearInterval(autoPlayBackward);
+  autoBackwardBtn.disabled = true;
+  autoForwardBtn.disabled = true;
+});
+
+stopBtn.addEventListener("click", () => {
+  clearInterval(autoPlayBackward);
+  clearInterval(autoPlayForward);
+  autoBackwardBtn.disabled = false;
+  autoForwardBtn.disabled = false;
+});
+
+window.onload = generatePhoto;
+
+// extra img navigate buttons
+
+const navContainer = document.createElement("div");
+navContainer.classList.add("nav");
+
+for (let i = 0; i < images.length; i++) {
+  const imgBtn = document.createElement("button");
+  imgBtn.classList.add(`img${i}`);
+  imgBtn.setAttribute("id", `img${i}`);
+  imgBtn.classList.add("navBtn");
+  navContainer.appendChild(imgBtn);
```

Comment:
Same here as I mentioned in the previous message, you created the element and then appended it to the DOM. Well done!